### PR TITLE
plugin/solidigm: Added vs-internal-log command.

### PR DESCRIPTION
**Pull request by lgdacunh**

### CODE REVIEW

**Comment by lgdacunh** [comment]
@igaw, "**apt install**" seems to be failing on "**build / cross armhf**" and "**build / cross s390x**". Can you please try to restart/fix this builds?

**Review by igaw — COMMENTED** [review]
If the armhf build doesn't pass till tomorrow I going to merge it anyway. Maybe we should start using the cross tool chain from bootlin anyway.

**Comment by igaw** [comment]
The armhf build found:

```
./plugins/solidigm/solidigm-internal-logs.c: In function ‘cmd_dump_repeat’:
../plugins/solidigm/solidigm-internal-logs.c:177:45: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
  177 |                         err = write(out_fd, (const void *)cmd->addr, cmd->data_len);
      |                                             ^
cc1: all warnings being treated as errors
```

So it's not the library install step which fails anymore

File: plugins/solidigm/solidigm-internal-logs.c [hunk 1]
```diff
@@ -0,0 +1,597 @@
...
+static int read_header(struct nvme_passthru_cmd *cmd, int ioctl_fd)
+{
+	memset((void *)(uintptr_t)cmd->addr, 0, INTERNAL_LOG_MAX_BYTE_TRANSFER);
+	return cmd_dump_repeat(cmd, INTERNAL_LOG_MAX_DWORD_TRANSFER, -1, ioctl_fd, false);
+}
+
+static int get_serial_number(char *str, int fd)
+{
+	struct nvme_id_ctrl ctrl = {0};
+	int err = 0;
```
[the remark this answers is not in the document]
Comment (lgdacunh) [reply plugins/solidigm/solidigm-internal-logs.c]: Done!

File: plugins/solidigm/solidigm-internal-logs.c [hunk 2]
```diff
@@ -0,0 +1,597 @@
...
+	struct assert_dump_header *ad = (struct assert_dump_header *) head_buf;
+	struct nvme_passthru_cmd cmd = {
+		.opcode = 0xd2,
+		.nsid = cfg.namespace_id,
+		.addr = (unsigned long)(void *)head_buf,
+		.cdw12 = ASSERTLOG,
+		.cdw13 = 0,
+	};
+	int output;
+	int err = read_header(&cmd, dev_fd(dev));
```
[the remark this answers is not in the document]
Comment (lgdacunh) [reply plugins/solidigm/solidigm-internal-logs.c]: Done!

File: plugins/solidigm/solidigm-internal-logs.c [hunk 3]
```diff
@@ -0,0 +1,597 @@
...
+	struct nvme_passthru_cmd cmd = {
+		.opcode = 0xd2,
+		.nsid = cfg.namespace_id,
+		.addr = (unsigned long)(void *)head_buf,
+		.cdw12 = EVENTLOG,
+		.cdw13 = 0,
+	};
+	int output;
+	int core_num;
+	int err = read_header(&cmd, dev_fd(dev));
```
[the remark this answers is not in the document]
Comment (lgdacunh) [reply plugins/solidigm/solidigm-internal-logs.c]: Done!

File: plugins/solidigm/solidigm-internal-logs.c [hunk 4]
```diff
@@ -0,0 +1,597 @@
...
+		if (nlog_header->ver.minor == 0)
+			return sizeof(struct nlog_dump_header4_0);
+		return sizeof(struct nlog_dump_header4_1);
+	default:
+		return INTERNAL_LOG_MAX_BYTE_TRANSFER;
+	}
+
+}
+
+// dumps nlogs from specified core or all cores when core = -1
```
[the remark this answers is not in the document]
Comment (lgdacunh) [reply plugins/solidigm/solidigm-internal-logs.c]: Done!

**Comment by lgdacunh** [comment]
@igaw , All checks passing now. All comments fixed.

**Comment by lgdacunh** [comment]
I am sorry for the branch deletion and restoration.
Some experiments went sideways when trying to enable the "Allow edits by maintainer" checkbox in GitHub.
The branch is back were it was with all the requested fixes in place, and all the checks passing.

**Comment by igaw** [comment]
No worries, luckily github is good in squashing notifications down in the ui (not using any email notifications :))

**Pull request merged**